variables.tf

```diff
@@ -0,0 +1,178 @@
+variable "create_iam_roles" {
+  type        = bool
+  description = "Flag to indicate whether to create IAM roles or not"
+  default     = true
+}
+
+variable "iam_roles_name_prefix" {
+  type        = string
+  description = "Name prefix for your EC2 role name and APP role name"
+  default     = ""
+}
+
+variable "ec2_role_name" {
+  type        = string
+  description = "EC2 role name"
+  default     = "aviatrix-role-ec2"
+}
+
+variable "app_role_name" {
+  type        = string
+  description = "APP role name"
+  default     = "aviatrix-role-app"
+}
+
+variable "external_controller_account_id" {
+  type        = string
+  description = "External controller account ID"
+  default     = ""
+}
+
+variable "secondary_account_ids" {
+  type        = list(string)
+  description = "Secondary account IDs"
+  default     = []
+}
+
+variable "availability_zone" {
+  type        = string
+  description = "Availability zone"
+  default     = ""
+}
+
+variable "vpc_cidr" {
+  type        = string
+  description = "VPC in which you want launch Aviatrix controller"
+  default     = "10.0.0.0/16"
+}
+
+variable "subnet_cidr" {
+  type        = string
+  description = "Subnet in which you want launch Aviatrix controller"
+  default     = "10.0.1.0/24"
+}
+
+variable "use_existing_vpc" {
+  type        = bool
+  description = "Flag to indicate whether to use an existing VPC"
+  default     = false
+}
+
+variable "vpc_id" {
+  type        = string
+  description = "VPC ID, required when use_existing_vpc is true"
+  default     = ""
+}
+
+variable "subnet_id" {
+  type        = string
+  description = "Subnet ID, required when use_existing_vpc is true"
+  default     = ""
+}
+
+variable "use_existing_keypair" {
+  type        = bool
+  description = "Flag to indicate whether to use an existing key pair"
+  default     = false
+}
+
+variable "keypair" {
```

Comment:
Perhaps we can make this optional? Auto generate a new key pair, if user does not explicitly provides one?